.githooks/pre-commit

```diff
@@ -1,4 +1,23 @@
 #!/usr/bin/env bash
 
-cargo make check-rs-format && \
-cargo make check-ts-format
+# Run check-rs-format if Rust source files were modified.
+git status | egrep "*.rs"
+if [ $? -eq 0 ]; then
+    cargo make check-rs-format
+    ret=$?
+    if [ $ret -ne 0 ]; then
+        exit $ret
+    fi
+fi
+
+# Run check-rs-format if Typescript source files were modified.
+git status | egrep "*.ts"
+if [ $? -eq 0 ]; then
+    cargo make check-ts-format
+    ret=$?
+    if [ $ret -ne 0 ]; then
+        exit $ret
```

Comment:
🙃 I don't think this `if` clause is needed because it's the last check you do.

Reply:
You are correct it is not needed.  I like the verbose format though, I feel it makes the code easier to read since the return value is so explicit there is no wasted thought spent considering what is returned.  I tend to do this in C code also for the same reason.